noir/noir-repo/compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_black_box.rs

```diff
@@ -205,6 +205,25 @@ pub(crate) fn convert_black_box_call(
                 )
             }
         }
+        BlackBoxFunc::VariableBaseScalarMul => {
+            if let (
+                [BrilligVariable::SingleAddr(point_x), BrilligVariable::SingleAddr(point_y), BrilligVariable::SingleAddr(scalar_low), BrilligVariable::SingleAddr(scalar_high)],
+                [BrilligVariable::BrilligArray(result_array)],
+            ) = (function_arguments, function_results)
+            {
+                brillig_context.black_box_op_instruction(BlackBoxOp::VariableBaseScalarMul {
+                    point_x: point_x.address,
+                    point_y: point_y.address,
+                    scalar_low: scalar_low.address,
+                    scalar_high: scalar_high.address,
+                    result: result_array.to_heap_array(),
+                });
+            } else {
+                unreachable!(
+                    "ICE: VariableBaseScalarMul expects one register argument and one array result"
+                )
+            }
+        }
```

Comment:
We should also add an integration test showing a variable base scalar mul being done in an unconstrained function to show this brillig gen is correct.

Reply:
Added the brillig test in https://github.com/AztecProtocol/aztec-packages/pull/6039/commits/b4c100967f509b160e7d173df6d79a03cf701866 .